rpc/grpc/src/main/java/com/spotify/heroic/rpc/grpc/GrpcRpcProtocol.java

```diff
@@ -215,7 +216,8 @@ public ClusterNode node() {
             }
 
             @Override
-            public AsyncFuture<FullQuery> query(final FullQuery.Request request) {
+            public AsyncFuture<FullQuery> query(final FullQuery.Request request, final Span span) {
+                // TODO: Do we have to use a span here with gRPC since its auto instrumented?
```

Comment:
We don't need to setup a span here, but we should make sure the parent installed in the thread-local context when [the client starts its own span](https://github.com/spotify/heroic/blob/b5ae95cc520f9f33a16f1c9ef3a4e0722587d44f/rpc/grpc/src/main/java/com/spotify/heroic/rpc/grpc/GrpcRpcClient.java#L80). Than can be done either here, or by removing the span from the signature and ensuring its set wherever `query` is called. 

I'd vote for the second option for simplicity.